fast_llm/layers/language_model/head.py

```diff
@@ -146,13 +144,16 @@ def forward(
     def _forward_backward(
         self, input_: torch.Tensor, kwargs: dict, losses: dict | None = None
     ) -> tuple[torch.Tensor, torch.Tensor | None]:
-        target = kwargs.get(
-            LanguageModelKwargs.labels
-            if self._config.distillation_model is None
-            else f"{self._config.distillation_model}_logits"
-        )
+        if kwargs.get(LanguageModelKwargs.labels) is None:
+            target = None
+        else:
+            target = kwargs.get(
+                LanguageModelKwargs.labels
+                if self._use_dpo_loss or self._config.distillation_model is None
+                else f"{self._config.distillation_model}_logits"
+            )
         if target is not None:
-            if self._config.distillation_model is None:
+            if self._config.distillation_model is None or self._use_dpo_loss:
```

Comment:
Checks for `self._use_dpo_loss` redundant since it doesn't support distillation?